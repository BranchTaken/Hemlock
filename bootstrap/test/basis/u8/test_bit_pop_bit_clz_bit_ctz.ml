open! Basis.Rudiments
open! Basis
open U8
open Format

let test () =
  printf "@[<h>";
  let rec test = function
    | [] -> ()
    | x :: xs' -> begin
        printf "bit_{pop,clz,ctz} %a -> %Lu, %Lu, %Lu\n"
          pp_x x (bit_pop x) (bit_clz x) (bit_ctz x);
        test xs'
      end
  in
  let xs = [
    kv 0L;
    kv 1L;
    kv 0x80L;
    kv 0xffL
  ] in
  test xs;
  printf "@]"

let _ = test ()
