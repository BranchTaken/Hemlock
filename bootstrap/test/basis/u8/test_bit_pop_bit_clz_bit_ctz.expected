bit_{pop,clz,ctz} 0x00u8 -> 0, 8, 8
bit_{pop,clz,ctz} 0x01u8 -> 1, 7, 0
bit_{pop,clz,ctz} 0x80u8 -> 1, 0, 7
bit_{pop,clz,ctz} 0xffu8 -> 8, 0, 0
